app/src/main/java/org/simple/clinic/mobius/MobiusDelegate.kt

```diff
@@ -0,0 +1,88 @@
+package org.simple.clinic.mobius
+
+import android.os.Bundle
+import android.os.Parcelable
+import com.spotify.mobius.Connectable
+import com.spotify.mobius.Connection
+import com.spotify.mobius.First
+import com.spotify.mobius.MobiusLoop
+import com.spotify.mobius.Next
+import com.spotify.mobius.android.MobiusAndroid
+import com.spotify.mobius.extras.Connectables
+import com.spotify.mobius.functions.Consumer
+import com.spotify.mobius.functions.Function
+import com.spotify.mobius.rx2.RxMobius
+import io.reactivex.ObservableTransformer
+import org.simple.clinic.crash.CrashReporter
+import org.simple.clinic.util.unsafeLazy
+
+class MobiusDelegate<M : Parcelable, E, F>(
+    private val defaultModel: M,
+    private val initFunction: (M) -> First<M, F>,
+    private val updateFunction: (M, E) -> Next<M, F>,
+    private val effectHandler: ObservableTransformer<F, E>,
+    private val modelUpdateListener: (M) -> Unit,
+    private val crashReporter: CrashReporter
+) : Connectable<M, E> {
+  private val modelKey = defaultModel::class.java.name
+  private val viewStateKey = "ViewState_$modelKey"
+
+  private lateinit var controller: MobiusLoop.Controller<M, E>
+
+  private var lastKnownModel: M? = null
+
+  private val loop by unsafeLazy {
+    RxMobius
+        .loop(
+            { model: M, event: E -> updateFunction(model, event) },
+            { effects -> effects.compose(effectHandler) }
+        )
+        .init(initFunction)
+        .eventSource(eventSource)
+  }
+
+  val eventSource by unsafeLazy {
```

Comment:
We should explicitly set the type to `EventSource<E>` and hide the concrete type.